src/server/system_services/account_server.js

```diff
@@ -1463,12 +1462,11 @@ function generate_access_keys() {
 }
 
 function verify_authorized_account(req) {
-    return P.resolve()
-        .then(() => bcrypt.compare(req.rpc_params.verification_password.unwrap(), req.account.password.unwrap()))
-        .then(match => {
-            if (!match) return false;
-            return true;
-        });
+    //operator connects by token and doesn't have the password property.
+    if (req.role === 'operator') {
+        return true;
+    }
+    return bcrypt.compare(req.rpc_params.verification_password.unwrap(), req.account.password.unwrap());
```

Comment:
ok. I took another look, and it seems that `bcrypt.compare` is indeed async and we should wait for (https://www.npmjs.com/package/bcrypt). Sorry for misleading you before.

it should be enough to just add async to the function declaration since you directly return the promise from `bcrypt.compare`